plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPickerBitbucketCloud.tsx

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useCallback, useEffect } from 'react';
+import { Field } from '@rjsf/core';
+import { useApi, Progress } from '@backstage/core';
+import { scaffolderApiRef } from '../../../api';
+import { useAsync } from 'react-use';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import Input from '@material-ui/core/Input';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import { BitbucketIntegration } from '@backstage/integration';
+
+function splitFormData(url: string | undefined) {
+  let host = undefined;
+  let workspace = undefined;
+  let project = undefined;
+  let repo = undefined;
+
+  try {
+    if (url) {
+      const parsed = new URL(`https://${url}`);
+      host = parsed.host;
+      workspace = parsed.searchParams.get('workspace') || undefined;
+      project = parsed.searchParams.get('project') || undefined;
+      repo = parsed.searchParams.get('repo') || undefined;
+    }
+  } catch {
+    /* ok */
+  }
+
+  return { host, workspace, project, repo };
+}
+
+function serializeFormData(data: {
+  host?: string;
+  workspace?: string;
+  project?: string;
+  repo?: string;
+}) {
+  if (!data.host) {
+    return undefined;
+  }
+  const params = new URLSearchParams();
+  if (data.workspace) {
+    params.set('workspace', data.workspace);
+  }
+  if (data.project) {
+    params.set('project', data.project);
+  }
+  if (data.repo) {
+    params.set('repo', data.repo);
+  }
+
+  return `${data.host}?${params.toString()}`;
+}
+
+export const RepoUrlPickerBitbucketCloud: Field = ({
```

Comment:
Do you think this could be worked into the existing `RepoUrlPicker` and have it select the input fields based on the integration type instead?